pkgs/build-support/mkshell/default.nix

```diff
@@ -28,18 +28,63 @@ let
     "propagatedNativeBuildInputs"
     "shellHook"
   ];
+
+  merged = {
+    buildInputs = mergeInputs "buildInputs";
+    nativeBuildInputs = packages ++ (mergeInputs "nativeBuildInputs");
+    propagatedBuildInputs = mergeInputs "propagatedBuildInputs";
+    propagatedNativeBuildInputs = mergeInputs "propagatedNativeBuildInputs";
+  };
+
+  docPathShellHook = let
+    unwrapCC = pkg: if pkg ? cc then unwrapCC pkg.cc else pkg;
+    docInputs = builtins.concatLists (builtins.attrValues merged)
+                # grab the unwrapped cc, as wrappers may or may not
+                # retain doc outputs
+                ++ (lib.optional (stdenv ? cc) (unwrapCC stdenv.cc))
+                ++ stdenv.defaultBuildInputs ++ stdenv.defaultNativeBuildInputs
+                ++ stdenv.initialPath;
+    docDrvs = lib.unique (builtins.filter lib.isDerivation docInputs);
+    docPaths = type: lib.pipe docDrvs [
+      (map (lib.getOutput type))
+      (map (out: builtins.toPath "${out}/share/${type}"))
+      (builtins.filter builtins.pathExists)
```

Comment:
These paths are not guaranteed to be realized at eval time even if they exist. Please filter at build time instead, if at all